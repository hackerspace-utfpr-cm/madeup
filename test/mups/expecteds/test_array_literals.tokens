1(0-0) ID: a
1(2-2) ASSIGN: =
1(4-4) LEFT_CURLY_BRACE: {
1(5-5) NEWLINE: \n
2(8-8) INTEGER: 1
2(9-9) COMMA: ,
2(10-10) NEWLINE: \n
3(13-13) INTEGER: 2
3(14-14) COMMA: ,
3(15-15) NEWLINE: \n
4(18-18) INTEGER: 3
4(19-19) NEWLINE: \n
5(20-20) RIGHT_CURLY_BRACE: }
5(21-21) NEWLINE: \n
6(22-26) ID: debug
6(28-28) ID: a
6(29-29) LEFT_BRACKET: [
6(30-30) INTEGER: 0
6(31-31) RIGHT_BRACKET: ]
6(33-34) EQUAL_TO: ==
6(36-36) INTEGER: 1
6(37-37) NEWLINE: \n
7(38-42) ID: debug
7(44-44) ID: a
7(45-45) LEFT_BRACKET: [
7(46-46) INTEGER: 1
7(47-47) RIGHT_BRACKET: ]
7(49-50) EQUAL_TO: ==
7(52-52) INTEGER: 2
7(53-53) NEWLINE: \n
8(54-58) ID: debug
8(60-60) ID: a
8(61-61) LEFT_BRACKET: [
8(62-62) INTEGER: 2
8(63-63) RIGHT_BRACKET: ]
8(65-66) EQUAL_TO: ==
8(68-68) INTEGER: 3
8(69-69) NEWLINE: \n
9(70-70) NEWLINE: \n
10(71-72) TO: to
10(74-81) ID: printAll
10(83-83) ID: l
10(84-84) NEWLINE: \n
11(87-91) ID: print
11(93-93) ID: l
11(94-94) NEWLINE: \n
12(95-97) END: end
12(98-98) NEWLINE: \n
13(99-99) NEWLINE: \n
14(100-107) ID: printAll
14(109-109) LEFT_CURLY_BRACE: {
14(110-110) INTEGER: 5
14(111-111) COMMA: ,
14(113-114) INTEGER: 10
14(115-115) COMMA: ,
14(117-117) NEWLINE: \n
15(118-119) INTEGER: 20
15(120-120) RIGHT_CURLY_BRACE: }
15(121-121) NEWLINE: \n
16(122-122) NEWLINE: \n
17(123-130) ID: printAll
17(132-132) LEFT_CURLY_BRACE: {
17(133-133) LEFT_CURLY_BRACE: {
17(134-134) INTEGER: 1
17(135-135) COMMA: ,
17(137-137) INTEGER: 2
17(138-138) COMMA: ,
17(140-140) INTEGER: 3
17(141-141) RIGHT_CURLY_BRACE: }
17(142-142) COMMA: ,
17(143-143) NEWLINE: \n
18(144-144) LEFT_CURLY_BRACE: {
18(145-145) INTEGER: 4
18(146-146) COMMA: ,
18(147-147) NEWLINE: \n
19(148-148) INTEGER: 5
19(149-149) COMMA: ,
19(150-150) NEWLINE: \n
20(151-151) INTEGER: 6
20(152-152) RIGHT_CURLY_BRACE: }
20(153-153) COMMA: ,
20(154-154) NEWLINE: \n
21(155-155) LEFT_CURLY_BRACE: {
21(156-156) INTEGER: 7
21(157-157) COMMA: ,
21(159-159) INTEGER: 8
21(160-160) COMMA: ,
21(161-161) NEWLINE: \n
22(162-162) INTEGER: 9
22(163-163) RIGHT_CURLY_BRACE: }
22(164-164) NEWLINE: \n
23(165-165) RIGHT_CURLY_BRACE: }
23(166-166) NEWLINE: \n
24(167-167) END_OF_FILE: END OF FILE
