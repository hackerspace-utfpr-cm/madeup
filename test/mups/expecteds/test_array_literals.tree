(block (define-variable a (ARRAYLITERAL (INTEGER 1) (INTEGER 2) (INTEGER 3))) (call debug (== (SUBSCRIPT (call-with-names a) (INTEGER 0)) (INTEGER 1))) (call debug (== (SUBSCRIPT (call-with-names a) (INTEGER 1)) (INTEGER 2))) (call debug (== (SUBSCRIPT (call-with-names a) (INTEGER 2)) (INTEGER 3))) (define printAll ((l)) (block (call print (call-with-names l)))) (call printAll (ARRAYLITERAL (INTEGER 5) (INTEGER 10) (INTEGER 20))) (call printAll (ARRAYLITERAL (ARRAYLITERAL (INTEGER 1) (INTEGER 2) (INTEGER 3)) (ARRAYLITERAL (INTEGER 4) (INTEGER 5) (INTEGER 6)) (ARRAYLITERAL (INTEGER 7) (INTEGER 8) (INTEGER 9)))))
