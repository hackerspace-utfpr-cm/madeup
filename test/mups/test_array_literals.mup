a = {
  1,
  2,
  3
}
debug a[0] == 1
debug a[1] == 2
debug a[2] == 3

to printAll l
  print l
end

printAll {5, 10, 
20}

printAll {{1, 2, 3},
{4,
5,
6},
{7, 8,
9}
}
