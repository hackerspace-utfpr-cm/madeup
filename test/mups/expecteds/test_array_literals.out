a[0] == 1: true
a[1] == 2: true
a[2] == 3: true
{5, 10, 20}
{{1, 2, 3}, {4, 5, 6}, {7, 8, 9}}
